packages/codec/README.md

```diff
@@ -1,15 +1,161 @@
-# Truffle Decoder Core
-This module provides interfaces for decoding Solidity variables.  This is a
-fairly low-level interface meant to be used by e.g. a debugger; for a
-higher-level interface, see the `truffle-contract-decoder` package instead.
+# Truffle Decoder
+
+This module provides interfaces for decoding contract state, transaction
+calldata, and events.
+
+It's split into three classes: The wire decoder, the contract decoder, and the
+contract instance decoder.  The wire decoder is associated to the project as a
+whole and decodes transaction calldata and events.  The contract decoder is
+associated to a specific contract class.  It has all the capabilities of the
+wire decoder, but in addition it acts as a factory for contract instance
+decoders.  The contract instance decoder is associated to a specific contract
+instance; it too has all the capabilities of the wire decoder, but it can also
+decode the state variables for the specific instance.  (In addition, in the case
+that the contract does not include a `deployedBytecode` field in its artifact,
+which can hinder decoding certain things, the contract instance decoder can
+sometimes work around this where the other decoders cannot.)
```

Comment:
Not sure there's really enough here for a table.  Also remember that, per your suggestion, the contract decoder doesn't actually decode things but rather just acts as a factory for contract instance decoders.  (Well, OK, it retains the wire decoder functions, but it doesn't add any new *decoding* functions, only factory stuff).

Instead of that, how about I just add some usage examples?  And then people can see the API docs for more.

Reply:
Tentatively I'm into that, but I'll have to see it. I'm just thinking to make the start of this document be as high-signal upfront as possible, and a table/list seemed like a way to achieve that.

Reply:
(captured this as note for follow-up issue)